File: yht0827/2월/2월 20일/카펫.java
class Solution {
	public int[] solution(int brown, int yellow) {
		int[] answer = new int[2];

		int carpet = brown + yellow;

		for (int i = 3; i <= carpet; i++) {
			int col = i;
			int row = carpet / col;

			if (row < 3) {
				continue;
			}

			if (row >= col) {
				if ((row - 2) * (col - 2) == yellow) {
					answer[0] = row;
					answer[1] = col;
					break;
				}
			}
		}
		return answer;
	}
}

File: zuzubibi/2월/2월 20일/카펫.py
def solution(brown, yellow):
    arr = []
    for i in range(1,int(yellow**(1/2))+1):
        if yellow % i ==0:
            arr.append([i,yellow//i])
    arr.sort(reverse=True)

    for col, row in arr:
        col, row = col +2, row + 2
        if (row+col-2)*2 == brown:
            return [row, col]

def anotherSolution(brown, yellow):
    for i in range(1, int(yellow**(1/2)+1)):
        if yellow % i == 0:
            if 2*(i + yellow//i) == brown-4:
                return [yellow//i+2, i+2]


print(solution(10,2)) # [4,3]
print(solution(8,1)) # [3,3]
print(solution(12,4))
print(solution(24,24)) # [8,6]